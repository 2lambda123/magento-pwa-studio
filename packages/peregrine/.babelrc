{
  "plugins": [
    "transform-object-rest-spread",
    "transform-class-properties",
    ["transform-react-jsx", { "pragma": "createElement" }],
    ["transform-runtime", {
      "helpers": true,
      "polyfill": false, // polyfills will be handled by preset-env
      "regenerator": false
    }]
  ],
  "presets": [
    ["env", {
      "targets": {
        "browsers": ["> 5%"]
      },
      "modules": false
    }]
  ],
  "env": {
    "test": {
      "plugins": [
        "transform-es2015-modules-commonjs",
        "transform-object-rest-spread",
        "transform-class-properties",
        ["transform-react-jsx", { "pragma": "createElement" }]
      ]
    }
  }
}
