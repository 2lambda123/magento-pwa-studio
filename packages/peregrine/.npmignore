.eslintrc.js
.storybook
/lib/**/__{docs,helpers,mocks,stories,tests}__/**
/scripts
