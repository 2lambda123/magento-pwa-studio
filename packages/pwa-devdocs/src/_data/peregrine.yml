title: Peregrine
entries:
  - label: Overview
    url: /peregrine/

  - label: Storybook UI testing
    url: /peregrine/testing/storybook/

  - label: Utility modules
    entries:

      - label: REST API client
        url: /peregrine/reference/rest-api-client/

  - label: React components
    entries:

      - label: ContainerChild
        url: /peregrine/reference/container-child/

      - label: Item
        url: /peregrine/reference/item/

      - label: Items
        url: /peregrine/reference/items/

      - label: List
        url: /peregrine/reference/list/

      - label: Router
        url: /peregrine/reference/router/
