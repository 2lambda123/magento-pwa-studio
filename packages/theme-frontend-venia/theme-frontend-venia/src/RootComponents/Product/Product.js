import { Component, createElement } from 'react';
import PropTypes from 'prop-types';

import classify from 'src/classify';
import Carousel from 'src/components/ProductImageCarousel';
import Options from 'src/components/ProductOptions';
import Quantity from 'src/components/ProductQuantity';
import RichText from 'src/components/RichText';
import mockData from './mockData';
import defaultClasses from './product.css';

class Product extends Component {
    static propTypes = {
        classes: PropTypes.shape({
            root: PropTypes.string
        }),
        data: PropTypes.shape({
            additionalInfo: PropTypes.string,
            description: PropTypes.string,
            images: PropTypes.arrayOf(
                PropTypes.shape({
                    id: PropTypes.string
                })
            ),
            name: PropTypes.string,
            options: PropTypes.arrayOf(
                PropTypes.shape({
                    id: PropTypes.string
                })
            ),
            price: PropTypes.string
        })
    };

    static defaultProps = {
        data: mockData
    };

    render() {
        const { classes, data } = this.props;

        return (
            <article className={classes.root}>
                <section className={classes.title}>
                    <h1 className={classes.productName}>
                        <span>{data.name}</span>
                    </h1>
                    <p className={classes.productPrice}>
                        <span>{data.price}</span>
                    </p>
                </section>
                <section className={classes.imageCarousel}>
                    <Carousel images={data.images} />
                </section>
                <section className={classes.actions}>
                    <button className={classes.action}>
                        <span>Add to Wishlist</span>
                    </button>
                </section>
                <section className={classes.options}>
                    <Options options={data.options} />
                </section>
                <section className={classes.quantity}>
                    <h2 className={classes.quantityTitle}>
                        <span>Quantity</span>
                    </h2>
                    <Quantity />
                </section>
                <section className={classes.cartActions}>
                    <button className={classes.addToCart}>
                        <span>Add to Cart</span>
                    </button>
                </section>
                <section className={classes.description}>
                    <h2 className={classes.descriptionTitle}>
                        <span>Product Description</span>
                    </h2>
                    <RichText content={data.description} />
                </section>
                <section className={classes.details}>
                    <h2 className={classes.detailsTitle}>
                        <span>Details</span>
                    </h2>
                    <RichText content={data.additionalInfo} />
                </section>
                <section className={classes.related}>
                    <h2 className={classes.relatedTitle}>
                        <span>Similar Products</span>
                    </h2>
                </section>
            </article>
        );
    }
}

export default classify(defaultClasses)(Product);
