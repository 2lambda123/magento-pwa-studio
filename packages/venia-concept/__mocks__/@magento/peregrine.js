import React from 'react';
import { BrowserPersistence } from './util/simplePersistence';

const peregrine = jest.requireActual('../../../peregrine/src');

// re-exports
const RestApi = {
    Magento2: {
        request: jest.fn()
    }
};

const Util = { BrowserPersistence };

// hooks
const useApolloContext = jest.fn(peregrine.useApolloContext);
const useDocumentListener = jest.fn(peregrine.useDocumentListener);
const useDropdown = jest.fn(peregrine.useDropdown);
const useQuery = jest.fn(peregrine.useQuery);
const useQueryResult = jest.fn(peregrine.useQueryResult);
const useSearchParam = jest.fn(peregrine.useSearchParam);

// components

/**
 * the Price component from @magento/peregrine
 * has browser-specific functionality and cannot
 * currently by rendered in the test environment
 */
const Price = jest.fn().mockReturnValue(<div />);

module.exports = {
    ...peregrine,
    Price,
    RestApi,
    Util,
    useApolloContext,
    useDocumentListener,
    useDropdown,
    useQuery,
    useQueryResult,
    useSearchParam
};
