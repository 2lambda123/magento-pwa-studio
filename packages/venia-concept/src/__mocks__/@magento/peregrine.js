import { createElement } from 'react';

/**
 * the Price component from @magento/peregrine
 * has browser-specific functionality and cannot
 * currently by rendered in the test environment
 */
export const Price = () => <div />;
