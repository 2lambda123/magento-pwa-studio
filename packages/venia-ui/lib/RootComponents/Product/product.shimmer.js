import React, { useMemo } from 'react';
import { string } from 'prop-types';
import Shimmer from '../../components/Shimmer';
import { BreadcrumbShimmer } from '../../components/Breadcrumbs';
import defaultClasses from '../../components/ProductFullDetail/productFullDetail.css';
import { useStyle } from '../../classify';

const ProductShimmer = (props) => {
    const { productType } = props;
    const classes = useStyle(defaultClasses, props.classes);

    const options = useMemo(() => {
        if (productType.includes('Configurable')) {
            // TODO return different shimmer structure
            return (
                <Shimmer />
            );
        }

        return null;
    }, [productType]);

    return (
        <Fragment>
            <BreadcrumbShimmer />
            <div className={classes.root}>
                <section className={classes.title}>
                    <h1 className={classes.productName}>
                        {productDetails.name}
                    </h1>
                    <p className={classes.productPrice}>
                        <Price
                            currencyCode={productDetails.price.currency}
                            value={productDetails.price.value}
                        />
                    </p>
                </section>
                <section className={classes.imageCarousel}>
                    <Carousel images={mediaGalleryEntries} />
                </section>
                <FormError
                    classes={{
                        root: classes.formErrors
                    }}
                    errors={errors.get('form') || []}
                />
                <section className={classes.options}>{options}</section>
                <section className={classes.quantity}>
                    <h2 className={classes.quantityTitle}>
                        <FormattedMessage
                            id={'global.quantity'}
                            defaultMessage={'Quantity'}
                        />
                    </h2>
                    <QuantityFields
                        classes={{ root: classes.quantityRoot }}
                        min={1}
                        message={errors.get('quantity')}
                    />
                </section>
                <section className={classes.actions}>
                    {cartActionContent}
                    <Suspense fallback={null}>
                        <WishlistButton {...wishlistButtonProps} />
                    </Suspense>
                </section>
                <section className={classes.description}>
                    <h2 className={classes.descriptionTitle}>
                        <FormattedMessage
                            id={'productFullDetail.productDescription'}
                            defaultMessage={'Product Description'}
                        />
                    </h2>
                    <RichText content={productDetails.description} />
                </section>
                <section className={classes.details}>
                    <h2 className={classes.detailsTitle}>
                        <FormattedMessage
                            id={'global.sku'}
                            defaultMessage={'SKU'}
                        />
                    </h2>
                    <strong>{productDetails.sku}</strong>
                </section>
            </div>
        </Fragment>
    );
};

ProductShimmer.propTypes = {
    productType: string.required
};

export default ProductShimmer;
