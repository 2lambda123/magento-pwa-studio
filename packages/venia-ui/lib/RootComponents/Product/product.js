import React, { Fragment } from 'react';
import { FormattedMessage } from 'react-intl';
import { useProduct } from '@magento/peregrine/lib/talons/RootComponents/Product/useProduct';

import ErrorView from '@magento/venia-ui/lib/components/ErrorView';
import { Title, Meta } from '@magento/venia-ui/lib/components/Head';
import { fullPageLoadingIndicator } from '@magento/venia-ui/lib/components/LoadingIndicator';
import ProductFullDetail from '@magento/venia-ui/lib/components/ProductFullDetail';
import mapProduct from '@magento/venia-ui/lib/util/mapProduct';

/*
 * As of this writing, there is no single Product query type in the M2.3 schema.
 * The recommended solution is to use filter criteria on a Products query.
 * However, the `id` argument is not supported. See
 * https://github.com/magento/graphql-ce/issues/86
 * TODO: Replace with a single product query when possible.
 */

const Product = () => {
    const talonProps = useProduct({
        mapProduct
    });

    const { error, loading, product } = talonProps;

    if (loading && !product) return fullPageLoadingIndicator;
    if (error && !product) return <ErrorView />;
    if (!product) {
        return (
            <h1>
                <FormattedMessage
                    id={'product.outOfStockTryAgain'}
                    defaultMessage={
                        'This Product is currently out of stock. Please try again later.'
                    }
                />
            </h1>
        );
    }

    // Note: STORE_NAME is injected by Webpack at build time.
    const title = `${product.name} - ${STORE_NAME}`;

    return (
        <Fragment>
            <Title>{title}</Title>
            <Meta name="description" content={product.meta_description} />
            <ProductFullDetail product={product} />
        </Fragment>
    );
};

export default Product;
