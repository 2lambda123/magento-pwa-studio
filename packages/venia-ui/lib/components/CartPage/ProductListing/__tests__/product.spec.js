import React from 'react';
import { act } from 'react-test-renderer';
import { createTestInstance } from '@magento/peregrine';

import Product from '../product';

jest.mock('../../../../classify');
jest.mock('@apollo/react-hooks', () => {
    const executeMutation = jest.fn(() => ({ error: null }));
    const useMutation = jest.fn(() => [executeMutation]);

    return { useMutation };
});

jest.mock('@magento/peregrine/lib/context/cart', () => {
    const state = { cartId: 'cart123' };
    const api = {};
    const useCartContext = jest.fn(() => [state, api]);

    return { useCartContext };
});

jest.mock('@magento/peregrine', () => {
    const Price = props => <span>{`$${props.value}`}</span>;

    return {
        ...jest.requireActual('@magento/peregrine'),
        Price
    };
});

const props = {
    item: {
        id: '123',
        product: {
            name: 'Unit Test Product',
            small_image: {
                url: 'unittest.jpg'
            }
        },
        prices: {
            price: {
                currency: 'USD',
                value: 100
            }
        },
        quantity: 1
    }
};

test('renders simple product correctly', () => {
    const tree = createTestInstance(<Product {...props} />);

    expect(tree.toJSON()).toMatchSnapshot();
});

test('renders mask on removal', () => {
    const propsWithClass = {
        ...props,
        classes: {
            root: 'root',
            mask: 'mask'
        }
    };
    const tree = createTestInstance(<Product {...propsWithClass} />);
    const { root } = tree;
    const { onClick } = root.findByProps({ text: 'Remove from cart' }).props;

    act(() => {
        onClick();
    });

    expect(tree.toJSON()).toMatchSnapshot();
});

test('renders configurable product correctly', () => {
    const configurableProps = {
        item: {
            ...props.item,
            configurable_options: [
                {
                    option_label: 'Option 1',
                    value_label: 'Value 1'
                },
                {
                    option_label: 'Option 2',
                    value_label: 'Value 2'
                }
            ]
        }
    };
    const tree = createTestInstance(<Product {...configurableProps} />);

    expect(tree.toJSON()).toMatchSnapshot();
});
