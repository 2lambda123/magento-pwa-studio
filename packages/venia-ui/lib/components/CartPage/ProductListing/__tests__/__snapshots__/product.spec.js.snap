// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders configurable product correctly 1`] = `
<li
  className="root "
>
  <div
    className="root container"
  >
    <img
      alt="Unit Test Product"
      className="image placeholder"
      loading="eager"
      src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAQAAAAFCAQAAADIpIVQAAAADklEQVR42mNkgAJGIhgAALQABsHyMOcAAAAASUVORK5CYII="
      width={100}
    />
    <img
      alt="Unit Test Product"
      className="image notLoaded"
      loading="lazy"
      onError={[Function]}
      onLoad={[Function]}
      sizes="100px"
      src="/media/catalog/product/unittest.jpg?auto=webp&format=pjpg&width=100&fit=cover"
      srcSet="/media/catalog/product/unittest.jpg?auto=webp&format=pjpg&width=40&height=50&fit=cover 40w,
/media/catalog/product/unittest.jpg?auto=webp&format=pjpg&width=80&height=100&fit=cover 80w,
/media/catalog/product/unittest.jpg?auto=webp&format=pjpg&width=160&height=200&fit=cover 160w,
/media/catalog/product/unittest.jpg?auto=webp&format=pjpg&width=320&height=400&fit=cover 320w,
/media/catalog/product/unittest.jpg?auto=webp&format=pjpg&width=640&height=800&fit=cover 640w,
/media/catalog/product/unittest.jpg?auto=webp&format=pjpg&width=960&height=1200&fit=cover 960w,
/media/catalog/product/unittest.jpg?auto=webp&format=pjpg&width=1280&height=1600&fit=cover 1280w,
/media/catalog/product/unittest.jpg?auto=webp&format=pjpg&width=1600&height=2000&fit=cover 1600w,
/media/catalog/product/unittest.jpg?auto=webp&format=pjpg&width=2560&height=3200&fit=cover 2560w"
      width={100}
    />
  </div>
  <div
    className="details"
  >
    <span
      className="name"
    >
      Unit Test Product
    </span>
    <dl
      className="options"
    >
      <dt
        className="optionLabel"
      >
        Option 1
         : 
        Value 1
      </dt>
      <dt
        className="optionLabel"
      >
        Option 2
         : 
        Value 2
      </dt>
    </dl>
    <span
      className="price"
    >
      <span>
        $100
      </span>
       ea.
    </span>
    <div
      className="quantity"
    >
      - 
      1
       +
    </div>
  </div>
  <div
    className="root"
  >
    <button
      className="kebab"
      onClick={[Function]}
    >
      <span
        className="root"
      >
        <svg
          fill="none"
          height="24"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="12"
            r="1"
          />
          <circle
            cx="12"
            cy="5"
            r="1"
          />
          <circle
            cx="12"
            cy="19"
            r="1"
          />
        </svg>
      </span>
    </button>
    <ul
      className="dropdown"
    >
      <li
        className="menuItem"
      >
        <button
          onMouseDown={[Function]}
        >
          <span
            className="root"
          >
            <svg
              fill="none"
              height="14px"
              stroke="rgb(var(--venia-teal))"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
              viewBox="0 0 24 24"
              width="14px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M20.84 4.61a5.5 5.5 0 0 0-7.78 0L12 5.67l-1.06-1.06a5.5 5.5 0 0 0-7.78 7.78l1.06 1.06L12 21.23l7.78-7.78 1.06-1.06a5.5 5.5 0 0 0 0-7.78z"
              />
            </svg>
          </span>
          <span
            className="text"
          >
            Move to favorites
          </span>
        </button>
      </li>
      <li
        className="menuItem"
      >
        <button
          onMouseDown={[Function]}
        >
          <span
            className="root"
          >
            <svg
              fill="none"
              height="14px"
              stroke="rgb(var(--venia-teal))"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
              viewBox="0 0 24 24"
              width="14px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M17 3a2.828 2.828 0 1 1 4 4L7.5 20.5 2 22l1.5-5.5L17 3z"
              />
            </svg>
          </span>
          <span
            className="text"
          >
            Edit item
          </span>
        </button>
      </li>
      <li
        className="menuItem"
      >
        <button
          onMouseDown={[Function]}
        >
          <span
            className="root"
          >
            <svg
              fill="none"
              height="14px"
              stroke="rgb(var(--venia-teal))"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
              viewBox="0 0 24 24"
              width="14px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <polyline
                points="3 6 5 6 21 6"
              />
              <path
                d="M19 6v14a2 2 0 0 1-2 2H7a2 2 0 0 1-2-2V6m3 0V4a2 2 0 0 1 2-2h4a2 2 0 0 1 2 2v2"
              />
            </svg>
          </span>
          <span
            className="text"
          >
            Remove from cart
          </span>
        </button>
      </li>
    </ul>
  </div>
</li>
`;

exports[`renders mask on removal 1`] = `
<li
  className="root mask"
>
  <div
    className="root container"
  >
    <img
      alt="Unit Test Product"
      className="image placeholder"
      loading="eager"
      src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAQAAAAFCAQAAADIpIVQAAAADklEQVR42mNkgAJGIhgAALQABsHyMOcAAAAASUVORK5CYII="
      width={100}
    />
    <img
      alt="Unit Test Product"
      className="image notLoaded"
      loading="lazy"
      onError={[Function]}
      onLoad={[Function]}
      sizes="100px"
      src="/media/catalog/product/unittest.jpg?auto=webp&format=pjpg&width=100&fit=cover"
      srcSet="/media/catalog/product/unittest.jpg?auto=webp&format=pjpg&width=40&height=50&fit=cover 40w,
/media/catalog/product/unittest.jpg?auto=webp&format=pjpg&width=80&height=100&fit=cover 80w,
/media/catalog/product/unittest.jpg?auto=webp&format=pjpg&width=160&height=200&fit=cover 160w,
/media/catalog/product/unittest.jpg?auto=webp&format=pjpg&width=320&height=400&fit=cover 320w,
/media/catalog/product/unittest.jpg?auto=webp&format=pjpg&width=640&height=800&fit=cover 640w,
/media/catalog/product/unittest.jpg?auto=webp&format=pjpg&width=960&height=1200&fit=cover 960w,
/media/catalog/product/unittest.jpg?auto=webp&format=pjpg&width=1280&height=1600&fit=cover 1280w,
/media/catalog/product/unittest.jpg?auto=webp&format=pjpg&width=1600&height=2000&fit=cover 1600w,
/media/catalog/product/unittest.jpg?auto=webp&format=pjpg&width=2560&height=3200&fit=cover 2560w"
      width={100}
    />
  </div>
  <div
    className="details"
  >
    <span
      className="name"
    >
      Unit Test Product
    </span>
    <span
      className="price"
    >
      <span>
        $100
      </span>
       ea.
    </span>
    <div
      className="quantity"
    >
      - 
      1
       +
    </div>
  </div>
  <div
    className="root"
  >
    <button
      className="kebab"
      onClick={[Function]}
    >
      <span
        className="root"
      >
        <svg
          fill="none"
          height="24"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="12"
            r="1"
          />
          <circle
            cx="12"
            cy="5"
            r="1"
          />
          <circle
            cx="12"
            cy="19"
            r="1"
          />
        </svg>
      </span>
    </button>
    <ul
      className="dropdown"
    >
      <li
        className="menuItem"
      >
        <button
          onMouseDown={[Function]}
        >
          <span
            className="root"
          >
            <svg
              fill="none"
              height="14px"
              stroke="rgb(var(--venia-teal))"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
              viewBox="0 0 24 24"
              width="14px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M20.84 4.61a5.5 5.5 0 0 0-7.78 0L12 5.67l-1.06-1.06a5.5 5.5 0 0 0-7.78 7.78l1.06 1.06L12 21.23l7.78-7.78 1.06-1.06a5.5 5.5 0 0 0 0-7.78z"
              />
            </svg>
          </span>
          <span
            className="text"
          >
            Move to favorites
          </span>
        </button>
      </li>
      <li
        className="menuItem"
      >
        <button
          onMouseDown={[Function]}
        >
          <span
            className="root"
          >
            <svg
              fill="none"
              height="14px"
              stroke="rgb(var(--venia-teal))"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
              viewBox="0 0 24 24"
              width="14px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M17 3a2.828 2.828 0 1 1 4 4L7.5 20.5 2 22l1.5-5.5L17 3z"
              />
            </svg>
          </span>
          <span
            className="text"
          >
            Edit item
          </span>
        </button>
      </li>
      <li
        className="menuItem"
      >
        <button
          onMouseDown={[Function]}
        >
          <span
            className="root"
          >
            <svg
              fill="none"
              height="14px"
              stroke="rgb(var(--venia-teal))"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
              viewBox="0 0 24 24"
              width="14px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <polyline
                points="3 6 5 6 21 6"
              />
              <path
                d="M19 6v14a2 2 0 0 1-2 2H7a2 2 0 0 1-2-2V6m3 0V4a2 2 0 0 1 2-2h4a2 2 0 0 1 2 2v2"
              />
            </svg>
          </span>
          <span
            className="text"
          >
            Remove from cart
          </span>
        </button>
      </li>
    </ul>
  </div>
</li>
`;

exports[`renders simple product correctly 1`] = `
<li
  className="root "
>
  <div
    className="root container"
  >
    <img
      alt="Unit Test Product"
      className="image placeholder"
      loading="eager"
      src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAQAAAAFCAQAAADIpIVQAAAADklEQVR42mNkgAJGIhgAALQABsHyMOcAAAAASUVORK5CYII="
      width={100}
    />
    <img
      alt="Unit Test Product"
      className="image notLoaded"
      loading="lazy"
      onError={[Function]}
      onLoad={[Function]}
      sizes="100px"
      src="/media/catalog/product/unittest.jpg?auto=webp&format=pjpg&width=100&fit=cover"
      srcSet="/media/catalog/product/unittest.jpg?auto=webp&format=pjpg&width=40&height=50&fit=cover 40w,
/media/catalog/product/unittest.jpg?auto=webp&format=pjpg&width=80&height=100&fit=cover 80w,
/media/catalog/product/unittest.jpg?auto=webp&format=pjpg&width=160&height=200&fit=cover 160w,
/media/catalog/product/unittest.jpg?auto=webp&format=pjpg&width=320&height=400&fit=cover 320w,
/media/catalog/product/unittest.jpg?auto=webp&format=pjpg&width=640&height=800&fit=cover 640w,
/media/catalog/product/unittest.jpg?auto=webp&format=pjpg&width=960&height=1200&fit=cover 960w,
/media/catalog/product/unittest.jpg?auto=webp&format=pjpg&width=1280&height=1600&fit=cover 1280w,
/media/catalog/product/unittest.jpg?auto=webp&format=pjpg&width=1600&height=2000&fit=cover 1600w,
/media/catalog/product/unittest.jpg?auto=webp&format=pjpg&width=2560&height=3200&fit=cover 2560w"
      width={100}
    />
  </div>
  <div
    className="details"
  >
    <span
      className="name"
    >
      Unit Test Product
    </span>
    <span
      className="price"
    >
      <span>
        $100
      </span>
       ea.
    </span>
    <div
      className="quantity"
    >
      - 
      1
       +
    </div>
  </div>
  <div
    className="root"
  >
    <button
      className="kebab"
      onClick={[Function]}
    >
      <span
        className="root"
      >
        <svg
          fill="none"
          height="24"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="12"
            r="1"
          />
          <circle
            cx="12"
            cy="5"
            r="1"
          />
          <circle
            cx="12"
            cy="19"
            r="1"
          />
        </svg>
      </span>
    </button>
    <ul
      className="dropdown"
    >
      <li
        className="menuItem"
      >
        <button
          onMouseDown={[Function]}
        >
          <span
            className="root"
          >
            <svg
              fill="none"
              height="14px"
              stroke="rgb(var(--venia-teal))"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
              viewBox="0 0 24 24"
              width="14px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M20.84 4.61a5.5 5.5 0 0 0-7.78 0L12 5.67l-1.06-1.06a5.5 5.5 0 0 0-7.78 7.78l1.06 1.06L12 21.23l7.78-7.78 1.06-1.06a5.5 5.5 0 0 0 0-7.78z"
              />
            </svg>
          </span>
          <span
            className="text"
          >
            Move to favorites
          </span>
        </button>
      </li>
      <li
        className="menuItem"
      >
        <button
          onMouseDown={[Function]}
        >
          <span
            className="root"
          >
            <svg
              fill="none"
              height="14px"
              stroke="rgb(var(--venia-teal))"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
              viewBox="0 0 24 24"
              width="14px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M17 3a2.828 2.828 0 1 1 4 4L7.5 20.5 2 22l1.5-5.5L17 3z"
              />
            </svg>
          </span>
          <span
            className="text"
          >
            Edit item
          </span>
        </button>
      </li>
      <li
        className="menuItem"
      >
        <button
          onMouseDown={[Function]}
        >
          <span
            className="root"
          >
            <svg
              fill="none"
              height="14px"
              stroke="rgb(var(--venia-teal))"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
              viewBox="0 0 24 24"
              width="14px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <polyline
                points="3 6 5 6 21 6"
              />
              <path
                d="M19 6v14a2 2 0 0 1-2 2H7a2 2 0 0 1-2-2V6m3 0V4a2 2 0 0 1 2-2h4a2 2 0 0 1 2 2v2"
              />
            </svg>
          </span>
          <span
            className="text"
          >
            Remove from cart
          </span>
        </button>
      </li>
    </ul>
  </div>
</li>
`;
